storage/innobase/log/log0log.cc

```diff
@@ -1044,12 +1044,13 @@ lsn_t log_t::write_buf() noexcept
         the current LSN are generated. */
         MEM_MAKE_DEFINED(buf + length, (write_size_1 + 1) - new_buf_free);
         buf[length]= 0; /* allow recovery to catch EOF faster */
+        if (UNIV_LIKELY_NULL(re_write_buf))
+          MEM_MAKE_DEFINED(re_write_buf + length, (write_size_1 + 1) -
+                           new_buf_free);
```

Comment:
I was a bit worried that this could cause a compiler warning, but I didn’t see that in any non-debug build log that I checked. So, let’s go for this; it should be the most readable solution.

Side note: `re_write_buf` is `nullptr` if and only if `resizing == NOT_RESIZING`. I’d expect the `if` to be optimized away in any optimized build, that is, starting with `-O1`.

Reply:
I'm all in favour of readable. Considering optimization in an infrequent action that includes large time items like filesystem writes seems a little futile.